app/models/services/locators/doi_paper_locator.rb

```
@@ -6,30 +6,79 @@ def column
   def find_paper
     return super if super
 
+    paper = Paper.new
+
+    crossref = Crossref.new
+    if (result = crossref.get_doi_metadata(self.locator_id))
```

Let's get rid of these if statements so that the whole process just explodes if the `get_doi_metadata` method fails.

---

same kind of idea as above.  The nil process is handling the "doi not in crossref" case.  I could do something more specific to the "resource not found" case.

---

If it really actually catastrophically fails, then we should let it fail. Otherwise, we should handle 4xx and 5xx responses from the external API gracefully: https://github.com/briankung/projectcredo/pull/99/files/421f3270169e3dadbce1f99dff2edfbee31b426c#r82536967